app/controllers/user/team_players_controller.rb

```diff
@@ -1,12 +1,33 @@
 class User::TeamPlayersController < ApplicationController
   before_action :require_user
+  skip_before_action :verify_authenticity_token, only: [:create]
+
 
   def destroy
-    team = current_user.teams.find(params[:team_id])
-    team_player = TeamPlayer.find_by(player_id: params[:id], team_id: params[:team_id])
+    team = current_user.teams.find(team_player_params[:team_id])
+    team_player = TeamPlayer.find_by(team_player_params)
     team_player.destroy
     flash[:success] = "Player deleted!"
     redirect_to user_team_path(team)
   end
 
+  def create
+    player = Player.find(team_player_params[:player_id])
+    if TeamPlayer.does_not_exist?(team_player_params)
+      TeamPlayer.create!(team_player_params)
```

Comment:
I used something like `team_player = TeamPlayer.find_or_initialize_by(team_player_params)`, then `if team_player.new_record?` (new_record? is a built-in method) followed by `team_player.save`. Just another option, but it might be simpler than the model class method.

Reply:
I like Tyler's idea. There are a lot of ActiveRecord methods I didn't know about.